packages/neos-ui/src/index.js

```diff
@@ -152,6 +152,29 @@ function * application() {
     });
 
     fetchWithErrorHandling.registerGeneralErrorHandler((message = 'unknown error') => {
+        // Check if the message is a Flow exception response
+        if (message.indexOf('Flow-Debug-Exception-Header') >= 0) {
+            const htmlContainer = document.createElement('div');
+            htmlContainer.innerHTML = message;
+            const exceptionHeader = htmlContainer.querySelector('.Flow-Debug-Exception-Header');
+            if (exceptionHeader && exceptionHeader.textContent) {
+                const exceptionSubject = exceptionHeader.querySelector('.ExceptionSubject');
+                message = exceptionSubject.textContent + ' - Check the logs for details';
+            } else {
+                message = 'Unknown error from unexpected HTML response. Check the logs for details.';
+            }
+        } else if (message.trim()[0] === '{') {
+            // Check if the message is a JSON string
+            try {
+                const jsonMessage = JSON.parse(message);
+                if (jsonMessage.error && jsonMessage.error.message) {
+                    message = jsonMessage.error.message + ' - Reference code "' + jsonMessage.error.referenceCode + '"';
```

Comment:
could have used a template literal here btw, but I don't mind to have it as is

Reply:
I'm not used to them and always forget...